host-configs/rzvernal-toss_4_x86_64_ib_cray-clang@16.0.0_hip.cmake

```diff
@@ -0,0 +1,119 @@
+#------------------------------------------------------------------------------
+# !!!! This is a generated file, edit at own risk !!!!
+#------------------------------------------------------------------------------
+# CMake executable path: /usr/tce/packages/cmake/cmake-3.21.1/bin/cmake
+#------------------------------------------------------------------------------
+
+#------------------------------------------------------------------------------
+# Compilers
+#------------------------------------------------------------------------------
+# Compiler Spec: clang@=16.0.0
+#------------------------------------------------------------------------------
+if(DEFINED ENV{SPACK_CC})
+
+  set(CMAKE_C_COMPILER "/usr/WS1/axom/libs/toss_4_x86_64_ib_cray/2023_06_21_15_15_50/spack/lib/spack/env/clang/clang" CACHE PATH "")
+
+  set(CMAKE_CXX_COMPILER "/usr/WS1/axom/libs/toss_4_x86_64_ib_cray/2023_06_21_15_15_50/spack/lib/spack/env/clang/clang++" CACHE PATH "")
+
+  set(CMAKE_Fortran_COMPILER "/usr/WS1/axom/libs/toss_4_x86_64_ib_cray/2023_06_21_15_15_50/spack/lib/spack/env/clang/flang" CACHE PATH "")
+
+else()
+
+  set(CMAKE_C_COMPILER "/opt/rocm-5.5.1/llvm/bin/amdclang" CACHE PATH "")
+
+  set(CMAKE_CXX_COMPILER "/opt/rocm-5.5.1/llvm/bin/amdclang++" CACHE PATH "")
+
+  set(CMAKE_Fortran_COMPILER "/opt/rocm-5.5.1/llvm/bin/amdflang" CACHE PATH "")
+
+endif()
+
+set(CMAKE_C_FLAGS "-Wno-int-conversion -O1" CACHE STRING "")
+
+set(CMAKE_CXX_FLAGS "-O1" CACHE STRING "")
+
+set(CMAKE_Fortran_FLAGS "-Mfreeform" CACHE STRING "")
+
+set(ENABLE_FORTRAN ON CACHE BOOL "")
+
+#------------------------------------------------------------------------------
+# MPI
+#------------------------------------------------------------------------------
+
+set(MPI_C_COMPILER "/usr/tce/packages/cray-mpich-tce/cray-mpich-8.1.25-rocmcc-5.5.1/bin/mpicc" CACHE PATH "")
+
+set(MPI_CXX_COMPILER "/usr/tce/packages/cray-mpich-tce/cray-mpich-8.1.25-rocmcc-5.5.1/bin/mpicxx" CACHE PATH "")
+
+set(MPI_Fortran_COMPILER "/usr/tce/packages/cray-mpich-tce/cray-mpich-8.1.25-rocmcc-5.5.1/bin/mpif90" CACHE PATH "")
+
+set(MPIEXEC_EXECUTABLE "/usr/bin/srun" CACHE PATH "")
+
+set(MPIEXEC_NUMPROC_FLAG "-n" CACHE STRING "")
+
+set(ENABLE_MPI ON CACHE BOOL "")
+
+#------------------------------------------------------------------------------
+# Hardware
+#------------------------------------------------------------------------------
+
+#------------------------------------------------------------------------------
+
+# HIP
+
+#------------------------------------------------------------------------------
+
+
+set(ENABLE_HIP ON CACHE BOOL "")
+
+set(HIP_ROOT_DIR "/opt/rocm-5.5.1/hip" CACHE STRING "")
+
+set(HIP_CLANG_INCLUDE_PATH "/opt/rocm-5.5.1/hip/../llvm/lib/clang/16.0.0/include" CACHE PATH "")
+
+set(CMAKE_CXX_FLAGS "--std=c++14" CACHE STRING "")
```

Comment:
Is this necessary? I thought BLT should handle this.

Also this is overriding the variable above.

Reply:
Good catch, this was a workaround for camp:
https://github.com/LLNL/axom/blob/2da4dd789acd2d46a70ec666a7ffb96b570581c2/scripts/spack/packages/axom/package.py#L291-L292

I'll test to see if it's still necessary, or define it just for camp if needed.

Reply:
I thought the preferred way was to use BLT_CXX_STD=c++14. @white238 which do you recommend.

Reply:
> I thought the preferred way was to use BLT_CXX_STD=c++14

We have this defined, up to BLT 0.6.1:
https://github.com/LLNL/axom/blob/2da4dd789acd2d46a70ec666a7ffb96b570581c2/scripts/spack/packages/axom/package.py#L233-L234

Not sure if forgot to update the version constraint here, or if BLT did change how it's set.

Reply:
Also the `cpp14` variant, that I put in, was a work around to get Sol working for Serac. I would prefer if it went away but oh well.